web/thesauruses/_meta/io.json

```diff
@@ -0,0 +1,243 @@
+            "deserialize_json_with_options"
+        ],
+        "XML": [
+            "serialize_xml",
+            "deserialize_xml",
+            "serialize_xml_with_options",
+            "deserialize_xml_with_options"
+        ]
+    },
+    "io": {
+        "write_line": {
+            "name": "Write a line to the console",
+            "code": ""
+        },
+        "write_line_with_new_line": {
+            "name": "Write a line to the console with a return charactor",
+            "code": ""
+        },
+        "read_line": {
+            "name": "Read a string of user input from the console",
+            "code": ""
+        },
+        "read_char": {
+            "name": "Read a single char of user input from the console",
+            "code": ""
+        },
+        "clear_console_cutput": {
+            "name": "Clear the console output",
+            "code": ""
+        },
+        "change_console_color": {
+            "name": "Change the color of the console",
+            "code": ""
+        },
+        "list_directory": {
+            "name": "List the items in the current directory",
+            "code": ""
+        },
+        "read_directory_info": {
+            "name": "Get infomation about the current directory (number of items, size, etc)",
+            "code": ""
+        },
+        "create_directory": {
+            "name": "Create a directory",
+            "code": ""
+        },
+        "delete_directory": {
+            "name": "Delete a directory",
+            "code": ""
+        },
+        "rename_directory": {
+            "name": "Rename a directory",
+            "code": ""
+        },
+        "move_directory": {
+            "name": "Move a directory to another locaiton",
+            "code": ""
+        },
+        "update_directory_permissions": {
+            "name": "Update the permissions a user, group, or application has to a directory",
+            "code": ""
+        },
+        "read_line_of_file": {
+            "name": "Read a single line from a file",
+            "code": ""
+        },
+        "read_lines_of_file": {
+            "name": "Read all the lines in a file",
+            "code": ""
+        },
+        "read_file_stream": {
+            "name": "Read the file into a stream",
+            "code": ""
+        },
+        "create_file": {
+            "name": "Create a file",
+            "code": ""
+        },
+        "delete_file": {
+            "name": "Delete a file",
+            "code": ""
+        },
+        "update_file": {
+            "name": "Update a file",
+            "code": ""
+        },
+        "move_file": {
+            "name": "Move a file to a new location",
+            "code": ""
+        },
+        "send_a_get_request": {
+            "name": "Send a HTTP GET request",
+            "code": ""
+        },
+        "send_a_post_request": {
+            "name": "Send a HTTP POST request",
+            "code": ""
+        },
```

Comment:
GET/POST are the most common, should we add PATCH/DELETE/whatever the other two are I don't remember off the top of my head?

Reply:
This is kind of tricky as alot of languages have a single 'Make http request' function which you pass the HTTP method into. Hence I went for Get and Post although I had originally also put a make http request section in which would cover the rest. Happy to do the 9 individually, just the most common plus a make http request, or just a make http request I'll let you have the preference :)

Reply:
Hmm... I guess for comparison's sake, it might make sense to have "make an HTTP request" and then the specific request types. More annoying to type out individually, but probably the most accurate.